lib/init.coffee

```diff
@@ -1,14 +1,52 @@
+{BufferedProcess, CompositeDisposable} = require 'atom'
+
 module.exports =
   config:
-    elixircExecutablePath:
-      type: 'string'
-      default: ''
-    includeDirs:
-      type: 'string'
-      default: ''
-    pa:
+    executablePath:
       type: 'string'
-      default: ''
+      title: 'Elixirc path'
+      default: '/usr/local/bin/elixirc'
 
   activate: ->
-    console.log 'activate linter-elixirc'
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.config.observe 'linter-elixirc.executablePath',
+      (executablePath) =>
+        @executablePath = executablePath
+  deactivate: ->
+    @subscriptions.dispose()
+  provideLinter: ->
+    helpers = require('atom-linter')
+    os = require 'os'
+    provider =
+      grammarScopes: ['source.elixir']
+      scope: 'file'
+      lintOnFly: false
+      name: 'Elixir'
+      lint: (textEditor) =>
+        filePath = textEditor.getPath()
+        elixirc_args = [
+          "--ignore-module-conflict", "--app", "mix", "--app", "ex_unit", "-o", os.tmpDir(),
+          filePath
+        ]
+        project_path = atom.project.getPaths()
+        opts =
+          cwd: project_path[0]
```

Comment:
How about you require `path` where you required `os` and then do `path.dirname(filePath)`